### ISSUE TITLE
Allow EntityRepository::findAll() method takes parameters for ordering

### ISSUE THREAD

**Issue by bocharsky-bw** [doctrine/doctrine2#5632]
I'm going to open a PR that allows ordering objects with [EntityRepository::findAll()](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/EntityRepository.php#L163-L166) method:
```php
    public function findAll(array $orderBy = null)
    {
        return $this->findBy(array(), $orderBy);
    }
```

I think `findAll()` method should be able to order entities because ordering causes no any effect for the  `WHERE`  clause.

What do you think about it?

**Comment by backbone87** [comment]
+1

**Comment by Ocramius** [comment]
Closing as "won't fix" as per #5652

**Comment by rdstudio** [comment]
Yes its ridiculous that you can't do a sort on findAll. Every other good programming language does this.
